speech/src/main/java/com/example/speech/InfiniteStreamRecognize.java

```diff
@@ -285,18 +215,121 @@ public void onError(Throwable t) {}
 
             tempByteString = ByteString.copyFrom(sharedQueue.take());
 
+            checkStopRecognitionFlag(tempByteString.toByteArray());
+
             request =
                 StreamingRecognizeRequest.newBuilder().setAudioContent(tempByteString).build();
 
             audioInput.add(tempByteString);
           }
-
+          Thread.sleep(100);
           clientStream.send(request);
         }
+        clientStream.closeSend();
       } catch (Exception e) {
         System.out.println(e);
       }
     }
   }
+
+  public static ResponseObserver<StreamingRecognizeResponse> getResponseObserver() {
+    return new ResponseObserver<StreamingRecognizeResponse>() {
+
+      final ArrayList<StreamingRecognizeResponse> responses = new ArrayList<>();
+
+      public void onStart(StreamController controller) {
+        referenceToStreamController = controller;
+      }
+
+      public void onResponse(StreamingRecognizeResponse response) {
+        responses.add(response);
+        StreamingRecognitionResult result = response.getResultsList().get(0);
+        Duration resultEndTime = result.getResultEndTime();
+        resultEndTimeInMS =
+                (int) ((resultEndTime.getSeconds() * 1000) + (resultEndTime.getNanos() / 1000000));
+        double correctedTime =
+                resultEndTimeInMS - bridgingOffset + (STREAMING_LIMIT * restartCounter);
+
+        SpeechRecognitionAlternative alternative = result.getAlternativesList().get(0);
+        if (result.getIsFinal()) {
+          System.out.print("\r");
+          System.out.printf(
+                  "%s: %s [confidence: %.2f]\n",
+                  convertMillisToDate(correctedTime),
+                  alternative.getTranscript(),
+                  alternative.getConfidence());
+          isFinalEndTime = resultEndTimeInMS;
+          lastTranscriptWasFinal = true;
+        } else {
+          System.out.print("\r");
+          System.out.printf(
+                  "%s: %s", convertMillisToDate(correctedTime), alternative.getTranscript());
+          lastTranscriptWasFinal = false;
+        }
+        checkStopRecognitionFlag(alternative.getTranscript().getBytes(StandardCharsets.UTF_8));
+      }
+
+      public void onComplete() {
+        System.out.println("Recognition was stopped");
+      }
+
+      public void onError(Throwable t) {}
+    };
+  }
+
+  private static void checkStopRecognitionFlag(byte[] flag) {
+    if (flag.length < 10) {
```

Comment:
nit: it would be nice to make more clear why `10` is selected as a threshold. consider using a constant with verbatim name or another method.